File: Stage30/Python/Problem2470.py
import sys
input = sys.stdin.readline

def main():
    N = int(input())
    arr = list(map(int, input().split()))
    
    arr.sort()
    
    left = 0
    right = N - 1
    
    ansL = arr[left]
    ansR = arr[right]
    min_val = float('inf')
    
    while left < right:
        sum_val = arr[left] + arr[right]
        
        if abs(sum_val) < min_val:
            min_val = abs(sum_val)
            ansL = arr[left]
            ansR = arr[right]
        
        if sum_val < 0:
            left += 1
        else:
            right -= 1
    
    print(ansL, ansR)

if __name__ == "__main__":
    main()

File: Stage30/Java/Problem2470.java
package Stage30.Java;

import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.util.Arrays;
import java.util.StringTokenizer;

public class Problem2470 {
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        StringBuilder sb = new StringBuilder();
        int N = Integer.parseInt(br.readLine());
        
        int[] arr = new int[N];
        StringTokenizer st = new StringTokenizer(br.readLine());
        
        for (int i = 0; i < N; i++) {
            arr[i] = Integer.parseInt(st.nextToken());
        }
        
        Arrays.sort(arr);
        
        int left = 0;
        int right = N - 1;
        
        int ansL = arr[left];
        int ansR = arr[right];
        long min = Long.MAX_VALUE;
        
        while (left < right) {
            long sum = (long) arr[left] + arr[right];
            
            if (Math.abs(sum) < min) {
                min = Math.abs(sum);
                ansL = arr[left];
                ansR = arr[right];
            }
            
            if (sum < 0) {
                left++;
            } else {
                right--;
            }
        }

        sb.append(ansL).append(" ").append(ansR);
        System.out.println(sb);
        br.close();
    }
}
